source/module_hsolver/diago_david.cpp

```diff
@@ -42,9 +42,9 @@ DiagoDavid<T, Device>::DiagoDavid(const Real* precondition_in,
     this->device = base_device::get_device_type<Device>(this->ctx);
     this->precondition = precondition_in;
 
-    this->one = &this->cs.one;
-    this->zero = &this->cs.zero;
-    this->neg_one = &this->cs.neg_one;
+    this->one = new T(static_cast<T>(1.0));
```

Comment:
is it possible to avoid using new and delete?

Reply:
I've changed constants to pointers to member objects to handle resources automatically.